health_monitoring/test_health_monitor.py

```diff
@@ -266,34 +183,28 @@ def test(self):
         """
         self.start_tempesta()
         
+        # 1
         back1 = self.get_server('nginx1')
         back2 = self.get_server('nginx2')
         back3 = self.get_server('nginx3')
-        res = self.run_curl(self.count)
-        time.sleep(2)
-        self.assertTrue(502 in res, "No 502 in statuses")
-        
-        print('start nginx1+2')
-        back1.run_start()
-        back2.run_start()
-        time.sleep(1)
-        # self.wait_for_server(back1)
-        res = self.run_curl(self.count)
-        self.assertTrue(502 in res and 404 in res, "Not valid status")
+        res = self.run_curl(REQ_COUNT)
+        self.assertTrue(list(set(res)) == [502], "No 502 in statuses")
         
-        print('start nginx1+2+3')
-        back3.run_start()
-        time.sleep(2)
-        res = self.run_curl(self.count)
-        self.assertTrue(200 in res, "Not valid status")
+        # 2
+        self.wait_for_server(back1)
+        self.wait_for_server(back2)
+        res = self.run_curl(REQ_COUNT)
+        self.assertTrue(sorted(list(set(res))) == [403, 404], "Not valid status")
+
+        # 3
+        self.wait_for_server(back3)
+        res = self.run_curl(REQ_COUNT)
+        self.assertTrue(sorted(list(set(res))) == [200, 403, 404, 502], "Not valid status")
         
-        print('stop nginx1+2')
-        back1.stop_nginx()
-        back2.stop_nginx()
-        time.sleep(2)
-        res = self.run_curl(self.count)
-        self.assertTrue(404 not in res, "Not valid status")
-        back3.stop_nginx()
+        # 4
+        res = self.run_curl(REQ_COUNT)
+        self.assertTrue(sorted(list(set(res))) == [200, 502], "Not valid status")
+        back3.stop()
 
 
 # class TestHealthMonitorCRCOnly(TestHealthMonitor):
```

Comment:
Need to delete, I believe

Reply:
Its another test with crc_check, if we agreed about this testcase i`ll complete it